library/src/main/java/com/mapzen/android/MapzenRouter.java

```diff
@@ -0,0 +1,174 @@
+package com.mapzen.android;
+
+import com.mapzen.valhalla.RouteCallback;
+import com.mapzen.valhalla.Router;
+import com.mapzen.valhalla.ValhallaRouter;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.util.Log;
+
+import java.util.HashMap;
+
+/**
+ * Main class for interaction with Mapzen's turn-by-turn service.
+ */
+public class MapzenRouter {
+
+  private static final String API_KEY_RES_NAME = "turn_by_turn_key";
+  private static final String API_KEY_RES_TYPE = "string";
+  private static final String TAG = MapzenRouter.class.getSimpleName();
+
+  private Router internalRouter = new ValhallaRouter();
+  private Context context;
+
+  private static final HashMap<DistanceUnits, Router.DistanceUnits> UNITS_TO_ROUTER_UNITS =
+      new HashMap<>();
+  static {
+    UNITS_TO_ROUTER_UNITS.put(DistanceUnits.MILES, Router.DistanceUnits.MILES);
+    UNITS_TO_ROUTER_UNITS.put(DistanceUnits.KILOMETERS, Router.DistanceUnits.KILOMETERS);
+  }
+
+  /**
+   * Creates a new {@link MapzenRouter}.
+   * @param context
+   */
+  public MapzenRouter(Context context) {
+    this.context = context;
+    initializeRouterKey();
+    initializeRouterUrl();
+  }
+
+  private void initializeRouterKey() {
+    final String packageName = context.getPackageName();
+    Resources res = context.getResources();
+    try {
+      final int apiKeyId = res.getIdentifier(API_KEY_RES_NAME, API_KEY_RES_TYPE, packageName);
+      final String apiKey = res.getString(apiKeyId);
+      internalRouter.setApiKey(apiKey);
```

Comment:
Eventually we will want to remove the concept of an API key param entirely from the on-the-road `ValhallaRouter` and replace it with a construct similar to the Tangram `HttpHandler` but I think this is fine for now.